internal/app/upstream/client.go

```diff
@@ -25,7 +25,7 @@ type Client interface {
 	// If the timeouts are exhausted, receive fails or a irrecoverable error occurs, the error is sent back to the caller.
 	// It is the caller's responsibility to send a new request from the last known DiscoveryRequest.
 	// Cancellation of the context cleans up all outstanding streams and releases all resources.
-	OpenStream(context.Context, *v2.DiscoveryRequest, string) chan *Response
+	OpenStream(context.Context, *v2.DiscoveryRequest) <-chan *Response
```

Comment:
Just so I understand, OpenStream is the one adding Responses to the receiver only channel? Thus the upstream/client.go has ownership of the `chan *Response` ?

Reply:
Yep, correct. upstream/client pushes responses to the receiver-only channel and has ownership of it.

@jyotimahapatra has a [WIP implementation](https://github.com/envoyproxy/xds-relay/pull/46) of it. Jyoti, I changed this to a receiver only channel so that orchestrator can read from it, let me know your thoughts as well.

Reply:
yeah. this is the intended behavior.